apps/server/src/modules/h5p-editor/uc/h5p.uc.ts

```diff
@@ -0,0 +1,335 @@
+
+			const result = await this.h5pAjaxEndpoint.postAjax(
+				query.action,
+				body,
+				language,
+				user,
+				filesFile && {
+					data: filesFile.buffer,
+					mimetype: filesFile.mimetype,
+					name: filesFile.originalname,
+					size: filesFile.size,
+				},
+				query.id,
+				undefined,
+				libraryUploadFile && {
+					data: libraryUploadFile.buffer,
+					mimetype: libraryUploadFile.mimetype,
+					name: libraryUploadFile.originalname,
+					size: libraryUploadFile.size,
+				},
+				undefined // TODO: HubID?
+			);
+
+			return result;
+		} catch (err) {
+			throw this.mapH5pError(err);
+		}
+	}
+
+	public async getContentParameters(contentId: string, currentUser: ICurrentUser) {
+		const user = this.changeUserType(currentUser);
+
+		try {
+			const result = await this.h5pAjaxEndpoint.getContentParameters(contentId, user);
+
+			return result;
+		} catch (err) {
+			throw new NotFoundException();
+		}
+	}
+
+	public async getContentFile(
+		contentId: string,
+		file: string,
+		req: Request,
+		currentUser: ICurrentUser
+	): Promise<{
+		data: Readable;
+		contentType: string;
+		contentLength: number;
+		contentRange?: { start: number; end: number };
+	}> {
+		const user = this.changeUserType(currentUser);
+
+		try {
+			const { mimetype, range, stats, stream } = await this.h5pAjaxEndpoint.getContentFile(
+				contentId,
+				file,
+				user,
+				this.getRange(req)
+			);
+
+			return {
+				data: stream,
+				contentType: mimetype,
+				contentLength: stats.size,
+				contentRange: range, // Range can be undefined, typings from @lumieducation/h5p-server are wrong
+			};
+		} catch (err) {
+			throw new NotFoundException();
+		}
+	}
+
+	public async getLibraryFile(ubername: string, file: string) {
+		try {
+			const { mimetype, size, stream } = await this.libraryService.getLibraryFile(ubername, file);
+
+			return {
+				data: stream,
+				contentType: mimetype,
+				contentLength: size,
+			};
+		} catch (err) {
+			throw new NotFoundException();
+		}
+	}
+
+	public async getTemporaryFile(
+		file: string,
+		req: Request,
+		currentUser: ICurrentUser
+	): Promise<{
+		data: Readable;
+		contentType: string;
+		contentLength: number;
+		contentRange?: { start: number; end: number };
+	}> {
+		const user = this.changeUserType(currentUser);
+
+		try {
+			const { mimetype, range, stats, stream } = await this.h5pAjaxEndpoint.getTemporaryFile(
+				file,
+				user,
+				// @ts-expect-error 2345: Callback can return undefined, typings from @lumieducation/h5p-server are wrong
+				this.getRange(req)
+			);
+
+			return {
+				data: stream,
+				contentType: mimetype,
+				contentLength: stats.size,
+				contentRange: range, // Range can be undefined, typings from @lumieducation/h5p-server are wrong
+			};
+		} catch (err) {
+			throw new NotFoundException();
+		}
+	}
+
+	public async getH5pPlayer(currentUser: ICurrentUser, contentId: string): Promise<IPlayerModel> {
+		// TODO: await this.checkPermission...
```

Comment:
You explained that permissions are a different PR, but it should of course be clear that the feature cant be released before this is done.